celements-config-source/src/main/java/org/xwiki/configuration/internal/AbstractDocumentConfigurationSource.java

```diff
@@ -0,0 +1,187 @@
+    return ((documentReference != null) && (classReference != null))
+        && (getDocumentAccessBridge().getProperty(documentReference, classReference, key) != null);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see ConfigurationSource#getKeys()
+   */
+  @Override
+  public List<String> getKeys() {
+    return null;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see ConfigurationSource#getProperty(String, Object)
+   */
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T> T getProperty(String key, T defaultValue) {
+    return getProperty(key, defaultValue, (Class<T>) defaultValue.getClass());
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see ConfigurationSource#getProperty(String, Class)
+   */
+  @Override
+  public <T> T getProperty(String key, Class<T> valueClass) {
+    T result = null;
+
+    DocumentReference documentReference = getFailsafeDocumentReference();
+    DocumentReference classReference = getFailsafeClassReference();
+    if ((documentReference != null) && (classReference != null)) {
+      result = (T) getDocumentAccessBridge().getProperty(documentReference, classReference, key);
+
+      // Make sure we don't return null values for List and Properties (they must return empty
+      // elements
+      // when using the typed API).
+      if (result == null) {
+        if (List.class.getName().equals(valueClass.getName())) {
+          result = (T) Collections.emptyList();
+        } else if (Properties.class.getName().equals(valueClass.getName())) {
+          result = (T) new Properties();
+        }
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see ConfigurationSource#getProperty(String)
+   */
+  @Override
+  public <T> T getProperty(String key) {
+    T result = null;
+    DocumentReference documentReference = getFailsafeDocumentReference();
+    DocumentReference classReference = getFailsafeClassReference();
+    if ((documentReference != null) && (classReference != null)) {
+      result = (T) getDocumentAccessBridge().getProperty(documentReference, classReference, key);
+    }
+
+    return result;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see ConfigurationSource#isEmpty()
+   */
+  @Override
+  public boolean isEmpty() {
+    return getKeys().isEmpty();
+  }
+
+  private <T> T getProperty(String key, T defaultValue, Class<T> valueClass) {
+    T result = getProperty(key, valueClass);
+    if (result == null) {
+      result = defaultValue;
+    }
+    return result;
+  }
+
+  private DocumentReference getFailsafeDocumentReference() {
+    DocumentReference documentReference;
+    try {
+      documentReference = getDocumentReference();
+    } catch (Exception e) {
+      // We verify that no error has happened and if one happened then we skip this configuration
+      // source. This
+      // ensures the system will continue to work even if this source has a problem.
+      documentReference = null;
```

Comment:
Please add logging.

Reply:
this PR's scope is forking & running static code formatting/cleanup. everything much beyond that is out of scope. we will have to adress things like these when we start changing these classes.

Reply:
I think that in the moment when we fork code, we make it to ower own code. This code therefore needs a proper review and quality improvments to level our code quality expectations. If done it for you, because you did not oppose the suggestion.

Reply:
@fpichler we shouldn't add more obstacles to the forking process, there is much left to fork and it's time-consuming enough already.
If we fork entirely new code I'd completely agree with you, but this is code we've been running since Celements 2 and it doesn't add new problems for us just by forking. A proper review and quality improvements can be done at any later stage.

Reply:
thanks for the changes